src/main/java/org/benetech/servicenet/repository/ActivityRepository.java

```diff
@@ -98,16 +102,34 @@ public ActivityRecord findOneByOrganizationId(UUID orgId) {
         return new PageImpl<>(results, pageable, total.intValue());
     }
 
-    private <T> void addFilters(CriteriaQuery<T> query, Root<ActivityInfo> root,
-        UUID ownerId, String searchName, FiltersActivityDTO filtersActivityDTO) {
+    private Predicate searchNames(Predicate predicate, From from, String searchName) {
+        return cb.and(predicate, cb.or(
+            cb.like(cb.upper(from.get(NAME)), '%' + searchName.trim().toUpperCase() + '%'),
+            cb.like(cb.upper(from.get(ALTERNATE_NAME)), '%' + searchName.trim().toUpperCase() + '%')
+        ));
+    }
+
+    private Predicate searchNames(Predicate predicate, Root<ActivityInfo> root, SearchOn searchOn, String searchName) {
+        if (searchOn.equals(SearchOn.ORGANIZATION)) {
+            return searchNames(predicate, root, searchName);
+        } else if (searchOn.equals(SearchOn.LOCATIONS)) {
+            Join<ActivityInfo, Organization> orgJoin = root.join(ORGANIZATIONS, JoinType.LEFT);
```

Comment:
in address filtering you will join the organization again, I think you can just make the join directly in addFilters method

Reply:
done